src/DeepCopy/DeepCopy.php

```diff
@@ -195,6 +196,10 @@ function ($object) {
                     }
                 );
 
+                if ($filter instanceof DoctrineProxyFilter) {
```

Comment:
$filterWasApplied is used to prevent recursive copy of property, if filter was applied. In other words you use filter or recursive copy. Removing $filterWasApplied breaks some fundamentals of DeepCopy.

Reply:
`$filterWasApplied` is only use when you use the `DoctrineProxyFilter` (=`ChainableFilter`) without any other filter (your use case).
I added this variable to keep the same behaviour (possible BC break) as before while fixing #98.

The issue before was that when you use the `DoctrineProxyFilter` with an other filter (`SetNullFilter` on the ID for example), the second filter was never used (so no ID with null value).
So, I added a condition to allow `ChainableFilter` (=`DoctrineProxyFilter`) to apply an other filter after (with the continue).
You can reach the `return` in the `if ($filterWasApplied) {` condition only if there is other filter to apply. There is already a `return` statement when you apply a filter.

Reply:
Got it now. You say, instead of adding "continue" on DoctrineProxyFilter, we need to remove $filterWasApplied variable. So that when chainable filter applied, either recursive copy or other filter applies after. Right?

Reply:
Yep :+1: (sorry if it wasn't clear the first time, english is not my native language :D)

Reply:
Done!